ios/build.sh

```diff
@@ -1,13 +1,29 @@
 #!/usr/bin/env bash
 
-set -e
-# set -x
-# set -u <-- difficult to test if variable is set this way...
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
 
-./kmbuild.sh $*
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
 
-if [ ! -z "$RELEASE_OEM" ]; then
-  pushd ../oem/firstvoices/ios > /dev/null
-  ./build.sh -lib-nobuild $*
-  popd > /dev/null
-fi
+# Include our resource functions; they're pretty useful!
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# Please note that this build script (understandably) assumes that it is running on Mac OS X.
+verify_on_mac
+
+builder_describe "Builds Keyman Engine and the Keyman app for use on iOS devices - iPhone and iPad." \
+  "clean" \
+  "configure" \
+  "build" \
+  ":engine          Builds KeymanEngine.xcframework, usable by our main app and by third-party apps" \
+  ":app=keyman      Builds the Keyman app for iOS platforms" \
+  "--debug+         Avoids codesigning and adds full sourcemaps for the embedded predictive-text engine" \
```

Comment:
possible TODO to add targets to Sample and OEM apps.

Reply:
Targets along the lines of 

https://github.com/keymanapp/keyman/blob/7b71e61b8aa2a5335f1424e24f038f2874b96f00/android/build.sh#L31-L36

Reply:
Note, the mismatch between iOS and Android for `:fv` is something we will need to address as well. While the iOS model _may_ be more elegant, the Android model is simpler :grin:

Reply:
It does make the `configure` step a bit simpler, to be sure:  KMW for the engine, keyboard resources for the app.

That said, this does likely make a difference for library consumers:  iOS provides `sil_euro_latin` as a default at all times, while this default is _not_ made available from the Android engine.  I can see pluses and minuses on both ends there - our keyboard-fallback logic is probably a bit _too_ simple if I recall, with it depending on a reliable default.

Reply:
> Targets along the lines of
> 
> https://github.com/keymanapp/keyman/blob/7b71e61b8aa2a5335f1424e24f038f2874b96f00/android/build.sh#L31-L36

... wait, does the `:fv` one even work as a "child project" reference?  My impression was that child projects had to be within subfolders of the parent script's folder.

Reply:
Note: the Android model of including the keyboard+model resources in the app rather than the engine is where we want to go long-term for both platforms. (Default fallback is a separate problem to solve -- what happens if sil_euro_latin goes wrong :grin:?)

Reply:
> ... wait, does the `:fv` one even work as a "child project" reference? My impression was that child projects had to be within subfolders of the parent script's folder.

Sneaky undocumented hack 🤣

Reply:
I've gone ahead and added them as targets for the main script and adjusted `ci.sh build` to only build `:engine` and `:app`, then selectively trigger the FV app.  After all, the samples have a separate build configuration, and the FV app still has its environment parameter flag.

As the samples' build config is set to use a different working directory, build-config backward-compatibility would be awkward if we changed that now.  

![image](https://github.com/keymanapp/keyman/assets/25213402/1e5d1a67-9bee-461d-ac18-17db63b36dfd)

The sample builds will continue to directly call the sample build.sh scripts at this time, as they're in the original working directory for their corresponding build step.  We can always change that once we hit 18.0 alpha / 17.0 stable and no longer need to directly call the 16.0 scripts.